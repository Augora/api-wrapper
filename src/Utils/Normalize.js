const { normalize } = require('normalizr');
const { depute } = require('./Schema/DeputesSchema');

function normalizeDeputes(normalizeFunction, deputesFromAPI) {
  return normalizeFunction(deputesFromAPI);
}

function normalizeDeputesFactory(normalizeFunction) {
  return normalizeDeputes.bind(undefined, normalizeFunction);
}

const providedNormalizeDeputes = normalizeDeputesFactory(data => normalize(data, [depute]));

module.exports = {
  normalizeDeputes,
  normalizeDeputesFactory,
  providedNormalizeDeputes,
};
