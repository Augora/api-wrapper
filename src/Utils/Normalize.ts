import { normalize } from "normalizr";
import { deputy } from "./Schema/DeputySchema";

export type Normalize = (o: object) => object;

const deputiesInOfficeNormalizer = (data: object) => normalize(data, [deputy]);

export function normalizeDeputes(
  normalizeFunction: Normalize,
  deputesFromAPI: object
) {
  return normalizeFunction(deputesFromAPI);
}

export function normalizeDeputesFactory(normalizeFunction: Normalize) {
  return normalizeDeputes.bind(undefined, normalizeFunction);
}

export const providedNormalizeDeputes = normalizeDeputesFactory(
  (data: object) => normalize(data, [deputy])
);

export function providedNormalizeDeputy(data: object) {
  return normalize(data, [deputy]);
}

export const providedNormalizeDeputesInOffice = normalizeDeputesFactory(
  deputiesInOfficeNormalizer
);
