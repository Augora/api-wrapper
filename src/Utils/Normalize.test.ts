import { normalizeDeputes, providedNormalizeDeputes } from "./Normalize";
import Seeds from "./DeputesSeeds";

test("calls the function", () => {
  const func = jest.fn();
  const data = {};
  normalizeDeputes(func, data);
  expect(func).toBeCalled();
});

test("return passed value", () => {
  const func = f => f;
  const data = {};
  expect(normalizeDeputes(func, data)).toBe(data);
});

test("Test deputy normalization with seed", () => {
  expect(providedNormalizeDeputes(Seeds[0])).toMatchSnapshot();
});

test("Test deputies normalization with seeds", () => {
  expect(providedNormalizeDeputes(Seeds)).toMatchSnapshot();
});
