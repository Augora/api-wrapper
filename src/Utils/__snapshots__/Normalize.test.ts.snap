// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test deputies normalization with seeds 1`] = `
Object {
  "entities": Object {
    "depute": Object {
      "1": Object {
        "date_naissance": "1956-02-02",
        "groupe_sigle": "LR",
        "id": 1,
        "id_an": "230329",
        "image120": "https://www.nosdeputes.fr/depute/photo/jean-francois-lamour/120",
        "image15": "https://www.nosdeputes.fr/depute/photo/jean-francois-lamour/15",
        "image30": "https://www.nosdeputes.fr/depute/photo/jean-francois-lamour/30",
        "image60": "https://www.nosdeputes.fr/depute/photo/jean-francois-lamour/60",
        "imageDynamic": [Function],
        "lieu_naissance": "Paris (Paris)",
        "mandat_debut": "2012-06-20",
        "nb_mandats": 2,
        "nom": "Jean-François Lamour",
        "nom_circo": "Paris",
        "nom_de_famille": "Lamour",
        "num_circo": 13,
        "num_deptmt": "75",
        "parti_ratt_financier": "Les Républicains",
        "place_en_hemicycle": "80",
        "prenom": "Jean-François",
        "profession": "Kinésithérapeute",
        "sexe": "H",
        "sites_web": Array [
          "http://www.jflamour.fr",
          "https://twitter.com/JF_Lamour",
        ],
        "slug": "jean-francois-lamour",
        "twitter": "JF_Lamour",
        "url_an": "http://www2.assemblee-nationale.fr/deputes/fiche/OMC_PA230329",
        "url_nosdeputes": "http://www.nosdeputes.fr/jean-francois-lamour",
        "url_nosdeputes_api": "http://www.nosdeputes.fr/jean-francois-lamour/json",
      },
      "2": Object {
        "date_naissance": "1954-10-31",
        "groupe_sigle": "SER",
        "id": 2,
        "id_an": "336439",
        "image120": "https://www.nosdeputes.fr/depute/photo/jean-michel-clement/120",
        "image15": "https://www.nosdeputes.fr/depute/photo/jean-michel-clement/15",
        "image30": "https://www.nosdeputes.fr/depute/photo/jean-michel-clement/30",
        "image60": "https://www.nosdeputes.fr/depute/photo/jean-michel-clement/60",
        "imageDynamic": [Function],
        "lieu_naissance": "Mauprevoir (Vienne)",
        "mandat_debut": "2012-06-20",
        "nb_mandats": 2,
        "nom": "Jean-Michel Clément",
        "nom_circo": "Vienne",
        "nom_de_famille": "Clément",
        "num_circo": 3,
        "num_deptmt": "86",
        "parti_ratt_financier": "Parti socialiste",
        "place_en_hemicycle": "263",
        "prenom": "Jean-Michel",
        "profession": "Avocat",
        "sexe": "H",
        "sites_web": Array [
          "https://twitter.com/jmclement86",
          "http://www.jeanmichelclement.com",
        ],
        "slug": "jean-michel-clement",
        "twitter": "jmclement86",
        "url_an": "http://www2.assemblee-nationale.fr/deputes/fiche/OMC_PA336439",
        "url_nosdeputes": "http://www.nosdeputes.fr/jean-michel-clement",
        "url_nosdeputes_api": "http://www.nosdeputes.fr/jean-michel-clement/json",
      },
    },
    "site": Object {
      "http://www.jeanmichelclement.com": Object {
        "site": "http://www.jeanmichelclement.com",
      },
      "http://www.jflamour.fr": Object {
        "site": "http://www.jflamour.fr",
      },
      "https://twitter.com/JF_Lamour": Object {
        "site": "https://twitter.com/JF_Lamour",
      },
      "https://twitter.com/jmclement86": Object {
        "site": "https://twitter.com/jmclement86",
      },
    },
  },
  "result": Array [
    1,
    2,
  ],
}
`;

exports[`Test deputy normalization with seed 1`] = `
Object {
  "entities": Object {
    "depute": Object {
      "undefined": Object {
        "0": Object {
          "site": "http://www.jflamour.fr",
        },
        "1": Object {
          "site": "https://twitter.com/JF_Lamour",
        },
        "image120": "https://www.nosdeputes.fr/depute/photo/undefined/120",
        "image15": "https://www.nosdeputes.fr/depute/photo/undefined/15",
        "image30": "https://www.nosdeputes.fr/depute/photo/undefined/30",
        "image60": "https://www.nosdeputes.fr/depute/photo/undefined/60",
        "imageDynamic": [Function],
      },
    },
  },
  "result": Array [
    1,
    "Jean-François Lamour",
    "Lamour",
    "Jean-François",
    "H",
    "1956-02-02",
    "Paris (Paris)",
    "75",
    "Paris",
    13,
    "2012-06-20",
    "LR",
    "Les Républicains",
    undefined,
    "Kinésithérapeute",
    "80",
    "http://www2.assemblee-nationale.fr/deputes/fiche/OMC_PA230329",
    "230329",
    "jean-francois-lamour",
    "http://www.nosdeputes.fr/jean-francois-lamour",
    "http://www.nosdeputes.fr/jean-francois-lamour/json",
    2,
    "JF_Lamour",
  ],
}
`;
